blueflood-http/src/main/java/com/rackspacecloud/blueflood/tracker/Tracker.java

```diff
@@ -40,12 +40,27 @@
     private static final String trackerName = String.format("com.rackspacecloud.blueflood.tracker:type=%s", Tracker.class.getSimpleName());
     private static final Pattern patternGetTid = Pattern.compile( "/v\\d+\\.\\d+/([^/]+)/.*" );
 
+    // Tracker is a singleton
+    private static Tracker instance = new Tracker();
+    private static boolean isRegistered = false;
+
     static Set tenantIds = new HashSet();
     static boolean isTrackingDelayedMetrics = false;
     static Set<String> metricNames = new HashSet<String>();
 
-    public Tracker() {
-        registerMBean();
+    // private constructor for singleton
+    private Tracker(){}
+
+    public static Tracker getInstance() {
+        register();
+        return instance;
+    }
+
+    public static void register() {
```

Comment:
@shintasmith 

I used singleton so that multiple servers classes (HttpMetricsIngestionServer & HttpMetricDataQueryServer) can use the same global object without having to create their own.
And I used static methods so that we don't have to pass the objects around to all the individual handlers.  I don't think they have to be mutually exclusive, and I think in this scenario it makes the code a lot simpler to read.  i.e. If I make these methods non-static, I'll have to grab the instance (with Tracker.getInstance()) for every single one of the handlers that needs it (HttpAggregatedIngestionHandler, HttpAggregatedMultiIngestionHandler, HttpEventsIngestionHandler, HttpMetricsIngestionHandler, HttpMultitenantMetricsIngestionHandler, HttpEventsQueryHandler, HttpHistogramQueryHandler, HttpMetricsIndexHandler, HttpMetricTokensHandler, HttpMultiRollupsQueryHandler, HttpRollupsQueryHandler), resulting to a lot more line of codes.  Remember this is more like a "logging" class, not a class that we have to make sure data are absolutely correct and atomic (where singletons *should* be retrieve by instance).  I think the tests setup we have for this class is sufficient (in TrackerTest), we don't need to mock a tracker class anywhere, even when testing the handlers code (similar to other singleton utilities such as Date and Loggers). 

I'll do it though if you think that's more correct.